aws/retry/middleware.go

```diff
@@ -30,13 +31,19 @@ type retryMetadataKey struct{}
 // AttemptMiddleware is a Smithy FinalizeMiddleware that handles retry attempts using the provided
 // Retryer implementation
 type AttemptMiddleware struct {
+	LogAttempts bool
+
 	retryer       Retryer
 	requestCloner RequestCloner
 }
 
 // NewAttemptMiddleware returns a new AttemptMiddleware
-func NewAttemptMiddleware(retryer Retryer, requestCloner RequestCloner) AttemptMiddleware {
-	return AttemptMiddleware{retryer: retryer, requestCloner: requestCloner}
+func NewAttemptMiddleware(retryer Retryer, requestCloner RequestCloner, optFns ...func(*AttemptMiddleware)) AttemptMiddleware {
+	m := AttemptMiddleware{retryer: retryer, requestCloner: requestCloner}
+	for _, fn := range optFns {
+		fn(&m)
+	}
+	return m
 }
```

Comment:
Does this `AttemptMiddleware` and its constructor need to be exported? Looks like its added via the `AddRetryMiddlewares`

Reply:
Let's track this separately as it does not pertain to this PR.